hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/format/cdc/CdcInputFormat.java

```diff
@@ -380,6 +382,7 @@ static class DataLogFileIterator implements ClosableIterator<RowData> {
           metaClient.getTableConfig().getPartialUpdateMode());
       this.logRecordIterator = logRecordIterator;
       initImages(cdcFileSplit);
+      this.deleteContext = new DeleteContext(props, tableSchema).withReaderSchema(tableSchema);
```

Comment:
seems for write path, the schemas we passed in are always the same, we can add a short-cut method like `DeleteContext.fromRecordSchema()` or something,

Reply:
I like this, I'll incorporate it once I get the tests passing